### PR TITLE
Use hadolint from pre-commit

### PR DESCRIPTION
**Pull request by mathbunnyru**
Thanks to https://github.com/hadolint/hadolint/pull/523 we can use hadolint directly in `pre-commit`.

Downsides:
- being able to run hadolint on a specific image with `make` command. But it is actually still possible to run it via `pre-commit --run` flags.
- need to be running docker to run hadolint. But this project uses docker for basically everything, so this should not be a problem.

Upsides:
- 21 deletions in Makefile
- approximately 50 deletions in docs
- The setup is easier because there are no specific things for hadolint
- No hadolint binary in HOME (I didn't like this on my mac)
- Pre-commit hooks versions could be updated automatically (I will try to do it later)

### CODE REVIEW

**Comment by mathbunnyru** [comment]
This is waiting for https://github.com/jupyter/docker-stacks/pull/1286 to be merged because there are same errors fixed in that PR and I don't want to have conflicts / do some work twice.

**Comment by mathbunnyru** [comment]
I also noted a bug in the hadolint-docker, which won't allow us to use it because it is better to fix the version of hadolint docker image.

https://github.com/hadolint/hadolint/issues/628

File: .pre-commit-config.yaml
Comment (romainx) [inline .pre-commit-config.yaml:9]:
```suggestion
    rev: v2.3.0
```

File: .pre-commit-config.yaml
Comment (romainx) [inline .pre-commit-config.yaml:13]:
```suggestion
        entry: hadolint/hadolint:v2.3.0 hadolint
```

File: .pre-commit-config.yaml
Comment (mathbunnyru) [reply .pre-commit-config.yaml:9]: I will merge master after your PR merge and fix the version in two places.
Unfortunately, right now we need to specify version in two places.
One - for revision of the hook, second - for the docker hadolint used.

**Review by romainx — CHANGES_REQUESTED** [review]
Hello,

Continuation of what we started together :smile:.
It's a good improvement and definitely the right solution.
I understand that for now we have to specify the version in two places, however it's not a big deal in comparison of all the stuff removed.

Thanks 🍰

**Review by romainx — APPROVED** [review]

**Comment by romainx** [comment]
Merged thanks a lot ❤️

**Pull request merged**
